src/core/pool_func.hpp

```diff
@@ -14,7 +14,7 @@
 #include "mem_func.hpp"
 #include "pool_type.hpp"
 
-extern void NORETURN SlErrorCorruptFmt(const char *format, ...) WARN_FORMAT(1, 2);
+#include "../saveload/saveload.h" // SlErrorCorrupt
```

Comment:
Maybe create a new `saveload_error.h` with the `SlError` functions? Including that in here will at least not include all the file IO stuff.